tests/api/test_api_demand_schedule.py

```diff
@@ -0,0 +1,66 @@
+import uuid
+
+import pytest
+
+TOKEN_HEADER = "bp2022-ap1-api-key"
+
+
+class TestApiColeDemandSchedule:
+    """
+    Test the /schedule/cole-demand endpoint
+    """
+
+    def test_get_all(self, client, clear_token):
+        response = client.get(
+            "/schedule/cole-demand", headers={TOKEN_HEADER: clear_token}
+        )
+        assert response.status_code == 200
+
+    @pytest.mark.parametrize(
+        "data",
+        [
+            {
+                "schedule_type": "TrainSchedule",
+                "demand_strategy_power_station": "power_station",
+                "demand_strategy_scaling_factor": 2,
+                "demand_strategy_start_datetime": "2022-01-01T00:00:00+00:00",
+                "platforms": ["platform1", "platform2"],
+            },
+            {
+                "schedule_type": "TrainSchedule",
+                "demand_strategy_power_station": "power_station",
+                "demand_strategy_scaling_factor": 2,
+                "demand_strategy_start_datetime": "2022-01-01T00:00:00+00:00",
+                "strategy_start_tick": 1,
+                "strategy_end_tick": 100,
+                "train_schedule_train_type": "passenger",
+                "platforms": ["platform1", "platform2"],
+            },
+        ],
+    )
+    def test_post(self, client, clear_token, data):
+        response = client.post(
+            "/schedule/cole-demand", headers={TOKEN_HEADER: clear_token}, json=data
+        )
+        assert response.status_code != 422
```

Comment:
Check for the right code not one wrong code :)

Reply:
The idea behind the `test_api_*` tests was to verify that the routes exist and validation works. The `test_impl_*` tests verify the functionality. That's the reason why I only check if the return codes don't equal 422.

Reply:
Ok, then go for it.